src/VisualStudio/Core/Def/Implementation/Diagnostics/VisualStudioWorkspaceDiagnosticAnalyzerProviderService.cs

```diff
@@ -34,15 +35,18 @@ internal partial class VisualStudioWorkspaceDiagnosticAnalyzerProviderService :
         private static readonly AnalyzerAssemblyLoader s_analyzerAssemblyLoader = new AnalyzerAssemblyLoader();
 
         [ImportingConstructor]
-        public VisualStudioWorkspaceDiagnosticAnalyzerProviderService(VisualStudioWorkspaceImpl workspace)
+        public VisualStudioWorkspaceDiagnosticAnalyzerProviderService(
+            [Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider)
         {
+            var dte = (EnvDTE.DTE)serviceProvider.GetService(typeof(EnvDTE.DTE));
```

Comment:
We don't have to fix every component, only ones in the transitive importing from any entrypoint that is a problem. If this was still being imported indirectly from VisualStudioWorkspace then yes, we'd still have problems.
Ideally, yes, we'd fix them all, but that's a very non-trivial piece of work and would be best aided by a time machine.